docs: Proofreading + general improvements (#26442)
renovatebot/renovate: docs/usage/getting-started/use-cases.md

Renovate:

1. Scans your repositories to find package files and their dependencies

> 2. Checks if any newer versions exist
> 3. Raises Pull Requests for available updates

Author: ```suggestion
1. Checks if any newer versions exist
1. Raises Pull Requests for available updates
```

A simplified example:

1. The repository has a `package.json` and `package-lock.json` with version `1.0.0` of a dependency

> 2. Renovate sees that version `1.1.0` is available
> 3. Renovate patches the `package.json` to change the dependency's version from `1.0.0` to `1.1.0`
> 4. Renovate runs `npm install` to let `npm` update the `package-lock.json`
> 5. Renovate commits the `package.json` and `package-lock.json`
> 6. Renovate creates the PR

Author: ```suggestion
1. Renovate sees that version `1.1.0` is available
1. Renovate patches the `package.json` to change the dependency's version from `1.0.0` to `1.1.0`
1. Renovate runs `npm install` to let `npm` update the `package-lock.json`
1. Renovate commits the `package.json` and `package-lock.json`
1. Renovate creates the PR
```

We split our work over two repositories:

1. The [`renovatebot/renovate`](https://github.com/renovatebot/renovate) repository, which has the Renovate code, and most of the Markdown documentation files

> 2. The [`renovatebot/renovatebot.github.io`](https://github.com/renovatebot/renovatebot.github.io) repository, which has a submodule link to the `renovatebot/renovate` repository

Author: ```suggestion
1. The [`renovatebot/renovatebot.github.io`](https://github.com/renovatebot/renovatebot.github.io) repository, which has a submodule link to the `renovatebot/renovate` repository
```

##### Update process

1. We edit our documentation files on the main Renovate repository `renovatebot/renovate`

> 2. Renovate sees the change(s) to the `renovatebot/renovate` Git submodule, and creates an update branch on the _documentation build_ repository
> 3. If the tests pass Renovate automerges the update branch into the `main` branch.
> 4. A GitHub Actions workflow runs on `main` to build the documentation site and push the build live on our GitHub Pages domain

Author: ```suggestion
1. Renovate sees the change(s) to the `renovatebot/renovate` Git submodule, and creates an update branch on the _documentation build_ repository
1. If the tests pass Renovate automerges the update branch into the `main` branch.
1. A GitHub Actions workflow runs on `main` to build the documentation site and push the build live on our GitHub Pages domain
```
